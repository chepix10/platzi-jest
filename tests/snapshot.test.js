import { getCharacter } from '../snapshot'
import rick from '../rick.json'

describe('Es hora de las instant치neas', () => {
  test('Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot()
  })

  test('Siempre fallar치 la instant치nea', () => {
    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 1000000)
    }
    expect(user).toMatchSnapshot()
  })

  test('Tenemos una excepci칩n', () => {
    const user = {
      id: Math.floor(Math.random() * 1000000),
      name: 'Oscar Barajas'
    }
    expect(user).toMatchSnapshot({
      id: expect.any(Number)
    })
  })
})
