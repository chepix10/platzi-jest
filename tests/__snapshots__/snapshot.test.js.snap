// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantáneas Siempre fallará la instantánea 1`] = `
Object {
  "createdAt": 2019-08-08T21:18:05.292Z,
  "id": 69507,
}
`;

exports[`Es hora de las instantáneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instantáneas Tenemos una excepción 1`] = `
Object {
  "id": Any<Number>,
  "name": "Oscar Barajas",
}
`;
